### DISCUSSION TITLE
Support for secure 433Mhz protocols

### DISCUSSION THREAD

**Discussion by gourry** [arendst/Tasmota#19770] [Q&A]
Hi all,
I've been using tasmota devices in multiple places for years.

Right now I'm dealing with a problem with my 2 garage and 1 gate doors. They rely on an obsolete "fixed code" schema that can be easily replicated by everybody. In fact it happened to me twice during the night that somebody opened the garage door and triggered the alarm. Luckily I was at home, but it was not fun at 3am in the morning.

Since both my garage doors and my front gate are already connected to an esp8266 running tasmota, I was investigating the possibility of simply getting rid of the built-in receiver and rely on Tasmota receiving commands from a 433Mhz remote.

However It's not clear to me if there is support for any form of encryption on this communication channel. Reading the documentation, I found that there is Keeloq support for shutters ( https://tasmota.github.io/docs/Blinds-and-Shutters/ ) but it looks like the support is to transmit and not to receive.

Can somebody tell me if there is a way to connect a secure 433Mhz remote to a device running Tasmota, and which type of remote/receiver would be needed?

Thank you in advance!

**Comment by Jason2866** [comment]
There is no support for RF 433 rolling codes in Tasmota. There are no plans to do.

**Reply by gourry** [reply]
Thanks!

**Comment by vic42** [comment]
If you buy a simple receiver module for your specific code, that outputs a voltage signal or relay contacts, such as a garage door receiver, you can connect that to a Tasmota switch input, without ever having to worry about the encryption. This works for all sorts of encrypted remotes.

**Reply by gourry** [reply]
Thanks, I actually started from this solution however it looked too pricy for what it is doing. Official receiver for my garage vendor is priced at 160€ ( https://www.telecomando-esma.com/ricevitore-nice-ox4t/ ) that is as expensive as replacing the control unit with a new one.
I looked at cheap knockoff on Aliexpress but didn't find a suitable one. For example this one looks legit at it has dip switches to select the vendor: https://www.aliexpress.com/item/32862609377.html
However it's only 2 channels and I need 4. I have a single antenna, and it would be very hard to add a second one, and looking on the internet they say that it's not possible to share an antenna accross 2 devices if they are both transmissing and receiving.
Than there are 4 channel solution like this: https://www.aliexpress.com/item/1005005922112989.html that I don't trust. They claim that they work on all the vendor, and they don't have a vendor selector. I suspect that they don't really support the rolling code packet. Since the button data is sent unencrypted they are probably just reading that data. In this case security would be identical to the one I have already with the fixed code.
Since security is really the concern from which I started this investigation, I'm overall not much convinced from the cheap obscure products.
Do you know about any other firmware that was built to support rolling code decoding? Foe example I saw a firmware called RFLink that mentions Keeloq support but on some threads they mentioned that decoding of the encrypted portion of the payload is not supported.

**Reply by vic42** [reply]
I'm not aware of any such firmware, which doesn't necessarily mean that it doesn't exist.
